scripts/ddt_library.py

```diff
@@ -259,31 +264,33 @@ def collect_ddt_fields(self, var_dict, var, run_env, ddt=None):
                 ctx = context_string(var.context)
                 errmsg = "Variable, {}, is not a known DDT{}"
                 raise ParseInternalError(errmsg.format(lname, ctx))
-            # End if
-        # End if
+            # end if
+        # end if
         for dvar in ddt.variable_list():
             subvar = VarDDT(dvar, var, self.run_env)
             dvtype = dvar.get_prop_value('type')
             if (dvar.is_ddt()) and (dvtype in self):
                 # If DDT in our library, we need to add sub-fields recursively.
                 subddt = self[dvtype]
                 self.collect_ddt_fields(var_dict, subvar, run_env, ddt=subddt)
-            else:
-                # add_variable only checks the current dictionary. For a
-                # DDT, the variable also cannot be in our parent dictionaries.
-                stdname = dvar.get_prop_value('standard_name')
-                pvar = var_dict.find_variable(standard_name=stdname,
-                                              any_scope=True)
-                if pvar:
-                    emsg = "Attempt to add duplicate DDT sub-variable, {}{}."
-                    emsg += "\nVariable originally defined{}"
-                    ntx = context_string(dvar.context)
-                    ctx = context_string(pvar.context)
-                    raise CCPPError(emsg.format(stdname, ntx, ctx))
-                # end if
-                # Add this intrinsic to <var_dict>
+            # end if
+            # add_variable only checks the current dictionary. By default,
+            # for a DDT, the variable also cannot be in our parent
+            # dictionaries.
+            stdname = dvar.get_prop_value('standard_name')
+            pvar = var_dict.find_variable(standard_name=stdname, any_scope=True)
+            if pvar and (not skip_duplicates):
+                ntx = context_string(dvar.context)
+                ctx = context_string(pvar.context)
+                emsg = f"Attempt to add duplicate DDT sub-variable, {stdname}{ntx}."
+                emsg += "\nVariable originally defined{ctx}"
```

Comment:
Doesn't this need to be an `f-string` as well? I guess we need to test our error messages better.

Reply:
good catch - fixed